### ISSUE TITLE
[Bug]: Contract deployment with Ledger blind signing off causes MetaMask to crash

### ISSUE THREAD

**Issue by plasmacorral** [MetaMask/metamask-mobile#11166]
### Describe the bug

Observed after update on LNX firmware 2.2.4 after updating to eth app ~1.11.2~ (EDIT: correction v1.11.3) that if I attempt to deploy a contract while blind signing is disabled, MetaMask v7.30.0 build 1410 will crash on both iOS 17.6.1 as well as android 14.

This was initially observed on feature branch for PR [10821](https://github.com/MetaMask/metamask-mobile/pull/10821) but confirmed to exist in production v7.30.0 build 1410

### Expected behavior

MetaMask should present an error or at least handle this situation more gracefully.

### Screenshots/Recordings

Screenshot of error before app close/crash:
<img width="200" alt="Screenshot 2024-09-11 at 8 01 04 PM" src="https://github.com/user-attachments/assets/2ae0fd49-5f57-42a0-b9a6-85809e3f090b">

[Recording](https://www.loom.com/share/a8a7909f540c4790bcdb29e5ba507990)

### Steps to reproduce

1. Have Ledger with Eth app 1.11.2 installed and have eth app settings on Ledger set to blind signing: disabled
2. Connect to [test dapp](https://metamask.github.io/test-dapp/) on MM mobile browser
3. Scroll down to ERC1155 section and tap "deploy"
4. Note "Confirm transaction on your Ledger" shown on mobile client
5. Note error "Unexpected error occurred. An unexpected error occurred.  Please try again"
6. Note MM closes on android and presents crash error on iOS

### Error messages or log output

```shell
Ledger device presents error: "Blind signing must be enabled in Settings"
```
In mobile client the following error is presented before app close/crash: 
```shell
"Unexpected error occurred. An unexpected error occurred.  Please try again"
```

### Version

7.30.0

### Build type

None

### Device

Pixel 5a and iPhone Xs

### Operating system

iOS, Android

### Severity

User can work around this by following suggestion on Ledger screen and enabling the setting within the Ledger Eth app.

**Comment by angelcheung22** [comment]
>Victor FORGEOUX
  Jul 26th at 4:35 PM
Hely Metamask team,
I’m Victor, managing the applications running on Ledger devices :wave:
We recently released an update of the Ethereum application (1.11.1 version) and while it shouldn’t have any impact on transactions signatures with Metamask, we’d like to request your help to modify some elements of your transaction modals.
The new Ethereum app version is a key component for our Clear Signing initiative and therefore :
The is no more blind signing setting in the app. Warning screens will be displayed before every transaction that can’t be decoded by the device (see attached visuals)
Users will still be able to transactions they used to sign, but they will have to review the risk on their device before doing so. The warning screens are automatically triggered and users don’t need to enable any app setting anymore without really understanding why they do it.
:point_right: The mentions about “blind signing” or “smart contract data” can be removed from Metamask Ledger interaction interfaces as they may be confusing for users from now on.

That means blind signing shall always be enabled automatically such that users dont need to enable it within the Eth app according to what they mentioned above, such that we also removed the 'enable eth dapp warning' per their request. 

We will follow up with Ledger on what is the issue here. @mikesposito

**Comment by vivek-consensys** [comment]
Tested on bitrise [build](https://app.bitrise.io/app/be69d4368ee7e86d/installable-artifacts/e9f4e40029fc0ec4/public-install-page/3719bf3a2458ffa5f346334553f44a0e)

Working as expected without any app crashes when blind signing is off.
